tests/subsys/logging/log_core_additional/src/log_test.c

```diff
@@ -425,6 +502,50 @@ void test_log_generic(void)
 	}
 }
 
+void test_log_msg2_create(void)
```

Comment:
what does this test validates? There is a `tests/subsys/logging/log_msg2` suite. does it miss something? It would be better to extend it then.

Reply:
run test suites in tests/subsys/logging/log_msg2 with "--coverage" can't cover z_log_msg2_static_create(). place these two cases here for
1. cover z_impl_z_log_msg2_static_create(), z_vrfy_z_log_msg2_static_create(), z_impl_z_log_msg2_runtime_vcreate(), z_vrfy_z_log_msg2_runtime_vcreate().
2. handle msg by backend defined in this file. z_log_msg2_claim() doesn't work well in user space.